app/views/vm_common/_snap.html.haml

```diff
@@ -1,40 +1,7 @@
-= form_tag({:action => 'snap_vm',
-            :id     => @record},
-           :id     => "vm_snap_form",
-           :method => :post) do
-  = render :partial => "layouts/flash_msg"
-  #form_div
-    %h3= _('Snapshot Information')
-    .form-horizontal
-      %div{:class => "form-group #{@record.vendor == "redhat" ? "hidden" : ""}"}
-        %label.control-label.col-md-2
-          = _('Name')
-        .col-md-8
-          = text_field_tag("name", @name, :maxlength => ViewHelper::MAX_NAME_LEN, :class => "form-control", "data-miq_focus" => true)
-      .form-group
-        %label.control-label.col-md-2
-          = _('Description')
-        .col-md-8
-          = text_area_tag("description", @description, :size => "50x4")
-      %div{:class => "form-group #{@show_snapshot_memory_checkbox ? "" : "hidden"}"}
-        %label.control-label.col-md-2
-          = _('Snapshot VM memory')
-        .col-md-8
-          = check_box_tag("snap_memory", "1", false)
-  - unless @edit
-    %hr
-    %table{:width => "100%"}
-      %tr
-        %td{:align => "right"}
-          = button_tag(_("Create"),
-            :name  => "create",
-            :class => "button",
-            :alt   => (t = _("Create snapshot")),
-            :title => t,
-            :type  => "submit")
-          = button_tag((t = _("Cancel")),
-            :name  => "cancel",
-            :class => "button",
-            :alt   => t,
-            :title => t,
-            :type  => "submit")
+= render :partial => "layouts/flash_msg"
+#snap-form{"data-create-url" => url_for_only_path(:action => "snap_vm", :id => @record.id, :button => @edit ? "create" : nil),
```

Comment:
I'll add it here again, I can't see value in using the dom as a place holder for data, we can avoid the extra dom lookup and putting data on it directly by simply passing json to the component factory.

was this done by design? /cc @himdel @vojtechszocs

Reply:
Does it matter? I mean, you either put the data here, and deal with the data types, escapement  with HAML and Ruby or you put the data in a hash, encode it in JSON and have it in the `:javascript` DOM node as code.

What's the benefit of one vs the other?

From the introspection perspective with this approach data stays data in the `data-*` attribute that is meant for data ;-) Is that wrong somehow?

@ohadlevy : I am trying to understand your concern so that we don't disregard your opinion. But I really don't see the problem. Neither does @himdel or @Hyperkid123.

From my perspective it would be best if the `#snap-form...` and the `Manageiq.component.componentFactory....` statements where a single entity. I'd see value in that. But if these are to be 2 pieces then I don't see much difference if the data is in the former or the latter.